packages/fxa-settings/src/components/ResetPasswordWarning/index.tsx

```diff
@@ -49,50 +58,80 @@ const ResetPasswordWarning = () => {
           aria-label={expanded ? 'Collapse warning' : 'Expand warning'}
         />
       </summary>
-      <div className="flex flex-col ps-8 pt-4 pb-2 gap-4">
-        <div className="flex items-start gap-2">
-          <IconSyncDevice
-            role="img"
-            className="flex-initial"
-            aria-hidden={true}
-          />
-          <div className="flex flex-col flex-1 -mt-1 gap-1">
-            <FtlMsg id="password-reset-previously-signed-in-device">
-              <p className="font-semibold">
-                Have a device where you previously signed in?
+      <div className="flex flex-col pt-4 pb-2 gap-4 text-xs leading-snug">
+        {locationState.recoveryKeyExists !== false && (
+          <div className="flex">
+            <IconSyncDevice
+              role="img"
+              className="me-4 mt-[2px]"
+              aria-hidden={true}
+            />
+            <div className="flex flex-col flex-1 gap-1">
+              <FtlMsg id="password-reset-warning-have-key">
+                <p className="font-semibold text-sm">
+                  Have an account recovery key?
+                </p>
+              </FtlMsg>
+              <FtlMsg id="password-reset-warning-use-key-link">
+                <Link
+                  to={`/account_recovery_confirm_key${searchParams}`}
+                  state={locationState}
+                  className="link-blue"
+                  onClick={() =>
+                    GleanMetrics.passwordReset.createNewRecoveryKeyMessageClick()
+                  }
+                >
+                  Use it now to reset your password and keep your data
+                </Link>
+              </FtlMsg>
+            </div>
+          </div>
+        )}
+        <div className="flex items-start gap-4">
+          <IconSyncDevice role="img" className="mt-[2px]" aria-hidden={true} />
+          <div className="flex flex-col flex-1 gap-1">
+            <FtlMsg id="password-reset-previously-signed-in-device-2">
+              <p className="font-semibold text-sm">
+                Have any device where you previously signed in?
+              </p>
+            </FtlMsg>
+            <FtlMsg id="password-reset-data-may-be-saved-locally-2">
+              <p className="text-grey-500">
```

Comment:
I double checked that this grey on the yellow passes AA and it does indeed for "normal text", not sure if there's a better tool to check text _this_ small though.